Sources/VergeCore/Inout.swift

```diff
@@ -0,0 +1,122 @@
+//
+// Copyright (c) 2020 muukii
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+
+@dynamicMemberLookup
+public final class Inout<State> {
+
+  public private(set) var hasModified = false
+
+  private let pointer: UnsafeMutablePointer<State>
+
+  public init(pointer: UnsafeMutablePointer<State>, onModified: @escaping () -> Void) {
```

Comment:
Passing the concrete instance causes a copy operation. 
To escape from this, use UnsafeMutablePointer.
I want to get behavior programmatically such as well as `inout` attribute in Swift language.

All of the methods to access the properties of the State use this pointer.

I don't know this works well and correctly.

Reply:
In addition, different from `inout` attribute, developers can do escaping `Inout` object to anywhere. 
this is the most dangerous scenario.

Reply:
`inout Inout<Wrapped>` probably works well? to prevent escaping in compile-time.

Reply:
If it work, I'll change the name of `Inout<Wrapped>` because it would be confusing.

Reply:
Changed `UnsafeInoutReference<Wrapped>`